builtin/credential/ldap/path_config.go

```diff
@@ -41,7 +43,25 @@ func pathConfig(b *backend) *framework.Path {
 
 			"groupdn": &framework.FieldSchema{
 				Type:        framework.TypeString,
-				Description: "LDAP domain to use for groups (eg: ou=Groups,dc=example,dc=org)",
+				Description: "LDAP search base to use for group membership search (eg: ou=Groups,dc=example,dc=org)",
+			},
+
+			"groupfilter": &framework.FieldSchema{
+				Type:    framework.TypeString,
+				Default: "(|(memberUid={{.Username}})(member={{.UserDN}})(uniqueMember={{.UserDN}}))",
+				Description: "Go template for querying group membership of user (optional)\n" +
+					"The template can access the following context variables: UserDN, Username\n" +
+					"Example: (&(objectClass=group)(member:1.2.840.113556.1.4.1941:={{.UserDN}}))\n" +
+					"Default: (|(memberUid={{.Username}})(member={{.UserDN}})(uniqueMember={{.UserDN}}))",
+			},
+
+			"groupattr": &framework.FieldSchema{
+				Type:    framework.TypeString,
+				Default: "cn",
+				Description: "LDAP attribute to follow on objects returned by <groupfilter>\n" +
+					"in order to enumerate user group membership.\n" +
+					"Examples: cn, memberOf\n" +
```

Comment:
Once you convert to backticks it should probably have the examples like

`Examples: "cn" or "memberOf"`

My main concern here is sometimes Vault accepts comma-separated lists, so some users may mistake this.